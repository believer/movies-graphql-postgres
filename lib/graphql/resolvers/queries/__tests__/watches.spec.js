const watches = require('../watches')
const { verifyToken } = require('../../../../services/token')

jest.mock('../../../../services/token', () => ({
  verifyToken: jest.fn().mockImplementation(() =>
    Promise.resolve({
      id: 2,
    })
  ),
}))

describe('queries/watches', () => {
  let db

  beforeEach(() => {
    db = {
      one: jest.fn().mockReturnValue('dbData'),
    }
  })

  it('should verify token', () => {
    watches(null, {}, { db, token: 'test' })

    expect(verifyToken).toHaveBeenCalledWith('test')
  })

  it('should get watch stats for a user', async () => {
    await watches(null, {}, { db })

    expect(db.one.mock.calls[0][0]).toMatchSnapshot()
  })

  it('should return database data', async () => {
    const response = await watches(null, {}, { db })

    expect(response).toEqual('dbData')
  })

  it('should handle errors', async () => {
    db.one.mockImplementation(() => Promise.reject('nope'))

    try {
      await watches(null, {}, { db })
    } catch (e) {
      expect(e.message).toMatch(/nope/)
    }
  })
})
