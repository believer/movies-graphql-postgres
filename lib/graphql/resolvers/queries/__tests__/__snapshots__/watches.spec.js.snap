// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries/watches should get watch stats for a user 1`] = `
"
      SELECT
        COUNT(distinct v.movie_id) AS total_views,
        COUNT(*) AS views_with_rewatches
      FROM views as v 
      WHERE v.user_id = 2
    "
`;
